src/main/java/org/jenkinsci/plugins/workflow/cps/global/UserDefinedGlobalVariableAction.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 Cisco Systems, Inc. and its affiliates
+ * Licensed under the MIT License
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.workflow.cps.global;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import hudson.model.AbstractBuild;
+import hudson.model.Api;
+import hudson.model.Run;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import jenkins.model.RunAction2;
+import org.jenkinsci.plugins.workflow.cps.global.UserDefinedGlobalVariable;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+
+import static hudson.Util.fixNull;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@ExportedBean(defaultVisibility = 999)
+public class UserDefinedGlobalVariableAction implements RunAction2 {
+
+    private transient Run run;
+    public List<UserDefinedGlobalVariable> userGlobalVars = new ArrayList<>();
```

Comment:
Right now this type cannot be serialized safely because `UserDefinedGlobalVariable.run` is non-transient. You can use my recommendation above to remove `UserDefinedGlobalVariable.run` entirely, but I would also recommend introducing a new inner class here that holds whatever properties are necessary so that future changes to `UserDefinedGlobalVariable` do not need to take serialization compatibility into account.